main.py

```diff
@@ -79,11 +86,25 @@ def main():
     params.timeout = args.timeout
     params.file = args.file
 
-    ### OSC Setup ###
-    use_old_reporter = args.use_old_reporter
+    ### Debug message toggle ###
+    if args.debug:
+        BoardShim.set_log_level(LogLevels.LEVEL_DEBUG.value)
+
+    ### Board Id selection ###
+    try:
+        master_board_id = int(args.board_id)
+    except ValueError:
+        master_board_id = BoardIds[args.board_id.upper()]
+    
+    ### Reporter Setup ###
     ip = args.osc_ip_address
     send_port = args.osc_port
-    osc_reporter = Old_OSC_Reporter(ip, send_port) if use_old_reporter else OSC_Reporter(ip, send_port)
+    use_old_reporter = args.use_old_reporter
+    reporters = [Old_OSC_Reporter(ip, send_port) if use_old_reporter else OSC_Reporter(ip, send_port)]
+    if args.debug:
+        reporters.append(Debug_Reporter(ip, send_port))
+    reporter = Reporter(reporters)
+    

```

Comment:
I think it'd also be a good idea to eventually break out all logic in the main function currently into smaller, focused functions which are all called via main. We don't need to do this for everything right now but since we're working on this part currently I think we should break this into a `setupReporter` function to start out.